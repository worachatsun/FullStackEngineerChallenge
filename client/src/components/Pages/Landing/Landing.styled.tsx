import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  top: 25%;
  position: fixed;
  display: flex;
  justify-content: center;
`;

export const Wrapper = styled.div`
  min-width: 50%;
  border-bottom: solid 8px #f03;
  box-shadow: 2px 0px 4px 0px rgba(0, 0, 0, 0.1);
`;

export const LogoDiv = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-around;
  font-size: 1.5em;
  margin-top: 50px;
`;

export const SigninDiv = styled.div`
  margin: 50px;
`;
