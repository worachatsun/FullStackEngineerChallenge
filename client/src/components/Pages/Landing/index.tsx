import React, { FunctionComponent } from "react";
import { useHistory } from "react-router-dom";
import Signin from "../Auth/Signin";
import { Container, LogoDiv, SigninDiv, Wrapper } from "./Landing.styled";

const Landing: FunctionComponent = () => {
    const history = useHistory();
    const token = localStorage.getItem("token");
    if (token) history.push("/employees");

    return (
        <Container>
            <Wrapper>
                <LogoDiv>
                    <img src="img_logo.png" alt="logo" />
                    <div>Performance Review</div>
                </LogoDiv>
                <SigninDiv>
                    <Signin />
                </SigninDiv>
            </Wrapper>
        </Container>
    );
};

export default Landing;
