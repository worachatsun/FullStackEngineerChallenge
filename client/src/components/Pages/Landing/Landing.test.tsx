import React from "react";
import renderer from "react-test-renderer";
import Landing from "./index";

describe("<Landing />", () => {
    test("should render correctly", () => {
        const render = renderer.create(<Landing />).toJSON();
        expect(render).toMatchSnapshot();
    });
});
