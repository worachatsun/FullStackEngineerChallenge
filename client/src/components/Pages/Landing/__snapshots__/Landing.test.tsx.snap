// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> should render correctly 1`] = `
<div
  className="sc-AxhCb GzWzT"
>
  <div
    className="sc-AxhUy iEcMhG"
  >
    <div
      className="sc-AxgMl bFtxng"
    >
      <img
        alt="logo"
        src="img_logo.png"
      />
      <div>
        Performance Review
      </div>
    </div>
    <div
      className="sc-AxheI gYMLrJ"
    >
      <div
        className="sc-AxiKw jAHgLp"
      >
        <form
          onSubmit={[Function]}
        >
          <div>
            <label
              htmlFor="Username"
            >
              Username
              :
              <input
                className="sc-AxirZ lmasuR"
                name="username"
                type="text"
              />
            </label>
          </div>
          <div>
            <label
              htmlFor="Password"
            >
              Password
              :
              <input
                className="sc-AxirZ lmasuR"
                name="password"
                type="password"
              />
            </label>
          </div>
          <button
            className="sc-AxjAm fbqdBC"
          >
            Signin
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
